---
title: "Summary plotting of metaBEAT output"
output: html_notebook
  toc: yes
---

#This markdown notebook outlines step by step the processing of [metaBEAT](https://github.com/HullUni-bioinformatics/metaBEAT) output data to produce the figures using in Kitson *et al.* (2017)

#Part 1: Minimum cluster depth analysis
```{r}
read.stats<-read.csv(file="data2/combined_read_stats.csv", stringsAsFactors=FALSE, header=TRUE)
read.stats<-subset(read.stats, clusters_min_cov<41)

ggplot(data = read.stats, aes(x=as.factor(clusters_min_cov), y=cluster_above_thres)) +
  geom_boxplot()
```


##Data processing
###First of all read in the libraries that we need.
```{r}
library(reshape2)
library(ggplot2)
```



###Next read in the data and metadata, we'll also trim the metadata to the bits we need.
```{r}
### read in the data - this is the read count output from metaBEAT, I have manually removed the .biom header line and the taxonomy column as this make live a lot easier in R
my.reads<-read.table(file="data2/metaBEAT-processed.tsv", stringsAsFactors=FALSE, header=TRUE,sep="\t")

### read in the sample by plate data - this is the querymap file used in metaBEAT with additional columns for PCR plate
my.plates<-read.csv(file="data2/sample_metadata.csv", stringsAsFactors=FALSE, header=TRUE)

### trim the plate data to the necessary columns
my.plates<-my.plates[, 1:3]

### greedy regex to split the sample string by the underscore and leave us with a column for nest and a column for indentifier
my.plates<-cbind(my.plates, do.call(rbind, strsplit(as.character(my.plates$sample), "_|_.*_")))

### trim the plate data to the necessary columns (i.e. drop the identifier column)
my.plates<-my.plates[, 1:4]
### name the columns
colnames(my.plates)<-c("sample", "plate", "plate.numeric", "nest")
```

###Merge the dataframes together using match and annotate the samples.
```{r}
### use match to add the plate data to the read data
my.reads$plate<-my.plates$plate[match(my.reads$sample, my.plates$sample)]
my.reads$plate.numeric<-my.plates$plate.numeric[match(my.reads$sample, my.plates$sample)]

### set the plotting order of the plates
my.reads$plate<-factor(reorder(my.reads$plate, my.reads$plate.numeric))

### make sample and plate factors for faceting and ordering
my.reads$sample<-as.factor(my.reads$sample)
my.reads$plate<-as.factor(my.reads$plate)

### set a grouping variable for the data type
my.reads$type<-ifelse(grepl("DNA", my.reads$sample), "DNApositive", 
                      ifelse(grepl("PCR", my.reads$sample), "PCRpositive", 
                             ifelse(grepl("Negative", my.reads$sample), "negative", "Moth sample")))

### subset the data to only keep the numbers of reads at each stage
my.reads.subs<-subset(my.reads, select=c("sample", "total", "trimmed.total", "queries", "plate", "type"))

### melt the data into long format
my.reads.subs.melt<-melt(my.reads.subs, id.vars=c("sample", "type", "plate"))
```

###Plot the read depth by plate including positives and negatives.
```{r}
### make the ggplot object + add the jittered dots + add the boxplots and colour them by trim level and make them a bit transparent
ggplot(aes(y = value, x = plate, fill = variable), data = my.reads.subs.melt) +
  ### make the boxplot and suppress the outliers as we are plotting the points anyway
  geom_boxplot(aes(fill=variable), alpha=0.5, position = position_dodge(width = 0.85), outlier.shape = NA) +
  ### plot the points
  geom_point(pch = 21, position = position_jitterdodge()) +
  ### fix the axes titles
  labs(y = "Reads per PCR well", x="PCR plate") +
  scale_fill_discrete(name="Trim level", 
                      breaks=c("total", "trimmed.total", "queries"), 
                      labels=c("Raw reads", "Trimmed reads", "Reads in clusters")) +
  scale_x_discrete(expand = c(0, 0)) +
  scale_y_continuous(expand = c(0, 0)) +
  ### rotate the x-axis labels and resize the text for the svg
  theme(axis.text.x = element_text(size=rel(1.5), colour="black", angle=45, hjust=1), 
        axis.text.y = element_text(size=rel(1.5), colour="black", angle=45, hjust=1), 
        axis.title.y = element_text(size=rel(2), vjust=2), 
        axis.title.x = element_text(size=rel(2), vjust=-1.3), 
        legend.text = element_text(size = rel(2.5)), 
        legend.title = element_text(size = rel(2.5), vjust=1), 
        legend.position = "right", 
        legend.key.height=unit(2, "line"), 
        legend.key=element_blank(), 
        legend.background = element_blank(), 
        panel.background = element_blank(), 
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(), 
        axis.line = element_blank(), 
        panel.border = element_rect(colour = "black", fill=NA, size=1), 
        plot.margin=unit(c(0.1, 0.1, 1, 1), "lines"))


### save the graph to an svg plot
ggsave(filename="diagrams/trimming_summary_all_samples.svg", width=300, height=300, units="mm")
```

###Plot the read depth by plate excluding positives and negatives.
```{r}
########################################################################################
################### Plot the data excluding the +ves and -ves  ###########################
########################################################################################

### make the ggplot object + add the jittered dots + add the boxplots and colour them by trim level and make them a bit transparent
ggplot(aes(y = value, x = plate, fill = variable), data = subset(my.reads.subs.melt, my.reads.subs.melt$type=="Moth sample")) +
  ### make the boxplot and suppress the outliers as we are plotting the points anyway
  geom_boxplot(aes(fill=variable), alpha=0.5, position = position_dodge(width = 0.85), outlier.shape = NA) +
  ### plot the points
  geom_point(pch = 21, position = position_jitterdodge()) +
  ### fix the axes titles
  labs(y = "Reads per PCR well", x="PCR plate") +
  scale_fill_discrete(name="Trim level", 
                      breaks=c("total", "trimmed.total", "queries"), 
                      labels=c("Raw reads", "Trimmed reads", "Reads in clusters")) +
  scale_x_discrete(expand = c(0, 0)) +
  scale_y_continuous(expand = c(0, 0)) +
  ### rotate the x-axis labels and resize the text for the svg
  theme(axis.text.x = element_text(size=rel(1.5), colour="black", angle=45, hjust=1), 
        axis.text.y = element_text(size=rel(1.5), colour="black", angle=45, hjust=1), 
        axis.title.y = element_text(size=rel(2), vjust=2), 
        axis.title.x = element_text(size=rel(2), vjust=-1.3), 
        legend.text = element_text(size = rel(2.5)), 
        legend.title = element_text(size = rel(2.5), vjust=1), 
        legend.position = "right", 
        legend.key.height=unit(2, "line"), 
        legend.key=element_blank(), 
        legend.background = element_blank(), 
        panel.background = element_blank(), 
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(), 
        axis.line = element_blank(), 
        panel.border = element_rect(colour = "black", fill=NA, size=1), 
        plot.margin=unit(c(0.1, 0.1, 1, 1), "lines"))

### save the graph to an svg plot
ggsave(filename="diagrams/trimming_summary_moths_only.svg", width=300, height=300, units="mm")
```


###Plot the read depth by PCR well type
```{r}
########################################################################################
################### Plot the data by sample type  ###########################
########################################################################################
### order the sample types for plotting
my.reads.subs.melt$type <- factor(my.reads.subs.melt$type, 
                                levels=c("Moth sample", 
                                         "DNApositive", 
                                         "PCRpositive", 
                                         "negative"))

### make the ggplot object + add the jittered dots + add the boxplots and colour them by trim level and make them a bit transparent
ggplot(aes(y = value, x = type, fill = variable), data = my.reads.subs.melt) +
  ### make the boxplot and suppress the outliers as we are plotting the points anyway
  geom_boxplot(aes(fill=variable), alpha=0.5, position = position_dodge(width = 0.85), outlier.shape = NA) +
  ### plot the points
  geom_point(pch = 21, position = position_jitterdodge()) +
  ### fix the axes titles
  labs(y = "Reads per PCR well", x="Sample type") +
  scale_fill_discrete(name="Trim level", 
                      breaks=c("total", "trimmed.total", "queries"), 
                      labels=c("Raw reads", "Trimmed reads", "Reads in clusters")) +
  scale_x_discrete(expand = c(0, 0)) +
  scale_y_continuous(expand = c(0, 0)) +
  ### rotate the x-axis labels and resize the text for the svg
  theme(axis.text.x = element_text(size=rel(1.5), colour="black", angle=45, hjust=1), 
        axis.text.y = element_text(size=rel(1.5), colour="black", angle=45, hjust=1), 
        axis.title.y = element_text(size=rel(2), vjust=2), 
        axis.title.x = element_text(size=rel(2), vjust=-1.3), 
        legend.text = element_text(size = rel(2.5)), 
        legend.title = element_text(size = rel(2.5), vjust=1), 
        legend.position = "right", 
        legend.key.height=unit(2, "line"), 
        legend.key=element_blank(), 
        legend.background = element_blank(), 
        panel.background = element_blank(), 
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(), 
        axis.line = element_blank(), 
        panel.border = element_rect(colour = "black", fill=NA, size=1), 
        plot.margin=unit(c(0.1, 0.1, 1, 1), "lines"))


### save the graph to an svg plot
ggsave(filename="diagrams/trimming_summary_sample_types.svg", width=300, height=300, units="mm")
```

