---
title: "Summary plotting of metaBEAT output"
output: html_notebook
---

#This markdown notebook outlines step by step the processing of [metaBEAT](https://github.com/HullUni-bioinformatics/metaBEAT) output data to produce the figures using in Kitson *et al.* (2017)

##Data processing
###First of all read in the libraries that we need.
```{r}
library(reshape2)
library(ggplot2)
```

###Next read in the data and metadata, we'll also trim the metadata to the bits we need.
```{r}
### read in the data - this is the read count output from metaBEAT, I have manually removed the .biom header line and the taxonomy column as this make live a lot easier in R
my.reads<-read.csv(file="../data2/minalilength__read_stats.csv", stringsAsFactors=FALSE, header=TRUE)

### read in the sample by plate data - this is the querymap file used in metaBEAT with additional columns for PCR plate
my.plates<-read.csv(file="../data2/sample_metadata.csv", stringsAsFactors=FALSE, header=TRUE)

### trim the plate data to the necessary columns
my.plates<-my.plates[,1:3]

### greedy regex to split the sample string by the underscore and leave us with a column for nest and a column for indentifier
my.plates<-cbind(my.plates, do.call(rbind, strsplit(as.character(my.plates$sample), "_|_.*_")))

### trim the plate data to the necessary columns (i.e. drop the identifier column)
my.plates<-my.plates[,1:4]
### name the columns
colnames(my.plates)<-c("sample","plate","plate.numeric","nest")
```

###Merge the dataframes together using match and annotate the samples.
```{r}
### use match to add the plate data to the read data
my.reads$plate<-my.plates$plate[match(my.reads$sample,my.plates$sample)]
my.reads$plate.numeric<-my.plates$plate.numeric[match(my.reads$sample,my.plates$sample)]

### set the plotting order of the plates
my.reads$plate<-factor(reorder(my.reads$plate, my.reads$plate.numeric))

### make sample and plate factors for faceting and ordering
my.reads$sample<-as.factor(my.reads$sample)
my.reads$plate<-as.factor(my.reads$plate)

### set a grouping variable for the data type
my.reads$type<-ifelse(grepl("DNA",my.reads$sample),"DNApositive",
                      ifelse(grepl("PCR",my.reads$sample),"PCRpositive",
                             ifelse(grepl("Negative",my.reads$sample),"negative","Moth sample")))
```
